### DISCUSSION TITLE
Use rofi just to display text?

### DISCUSSION THREAD

**Discussion by mil-ad** [davatorium/rofi#2065] [Q&A]
Is it possible to use rofi to just display some formatted text without any lists or matching? There is `rofi -e "foo"` but that seems to be just for one-liners.

**Comment by DaveDavenport** [comment]
You can do multi-line just fine with -e.

```bash
 rofi -e "Why is this just
for one liners,
when you can do multiple lines easy" -no-config

```

![rofi-2024-12-31-1439-00000](https://github.com/user-attachments/assets/8b00b327-8bd6-4452-8b93-b1204c3ca8d7)

**Comment by mil-ad** [comment]
Ah thanks! I was using double quotes with `echo`:

```
rofi -e $(echo "Why is this just\nfor one liners\nwhen you can do multiple lines easy")
```

but it needs single-quotes or `printf`

**Comment by mil-ad** [comment]
I think another difference is that with a pipe I can do streaming text but with `rofi -e "..."` it has to wait for the whole text to appear?

e.g. 

```
rofi -e "$(printf 'foo'; sleep 10; printf '\nbar')"
```

as opposed to

```
(printf 'foo'; sleep 10; printf '\nbar') | rofi -dmenu
```

**Reply by DaveDavenport** [reply]
If you run latest rofi you can also do:

```bash
echo -en "Why is this just\nfor one liners,\nwhen you can do multiple lines easy" | rofi -no-config -e -
```

**Reply by DaveDavenport** [reply]
https://davatorium.github.io/rofi/1.7.6/rofi.1/#message-dialog

**Reply by mil-ad** [reply]
ah very nice! 1.7.6 hasn't landed in Arch repos yet but hopefully soon 🤞🏼
